src/libmongoc/tests/test-mongoc-error.c

```diff
@@ -133,6 +134,52 @@ test_has_label (void)
    BSON_ASSERT (!mongoc_error_has_label (tmp_bson ("{}"), "foo"));
 }
 
+static void
+test_state_change_helper (uint32_t domain, bool expect_error)
+{
+   bson_error_t error;
+   uint32_t not_master_codes[] = {10107, 13435};
+   uint32_t node_is_recovering_codes[] = {11600, 11602, 13436, 189, 91};
+   uint32_t shutdown_codes[] = {11600, 91};
+   int i;
+
+   MONGOC_DEBUG ("Checking domain = %d", domain);
+
+   memset (&error, 0, sizeof (bson_error_t));
+   error.domain = domain;
+
+   for (i = 0; i < sizeof (not_master_codes) / sizeof (uint32_t); i++) {
+      error.code = not_master_codes[i];
+      BSON_ASSERT (expect_error == _mongoc_error_is_not_master (&error));
+      BSON_ASSERT (expect_error == _mongoc_error_is_state_change (&error));
+   }
+   for (i = 0; i < sizeof (node_is_recovering_codes) / sizeof (uint32_t); i++) {
+      error.code = node_is_recovering_codes[i];
+      BSON_ASSERT (expect_error == _mongoc_error_is_state_change (&error));
```

Comment:
```suggestion
      BSON_ASSERT (expect_error == _mongoc_error_is_state_change (&error));
      BSON_ASSERT (expect_error == _error_is_recovering (&error));
```
Was leaving out testing for `_error_is_recovering` intentional?

Reply:
I omitted it because it was static and not exposed outside of mongoc-error.c. Though I don't see much harm in adding to the private header for the tests. Done.